### ISSUE TITLE
Supporting optional parameters

### ISSUE THREAD

**Issue by slifty** [joeandaverde/tinypg#43]
This might be a terrible idea but I'm raising it just in case!

Would it be possible to provide support for optional parameters in `.sql` files?  For instance maybe `:id?` would create a parameter that will not error when a value is omitted.

My use case is that I'm looking into creating a single `updateById.sql` file for one of my entities, and I'd like to allow calling code to only pass the fields that need to be updated.  I'd be constructing the query itself to omit updates to fields with no value, for instance using something like:

```
UPDATE some_table SET
  column_1 = COALESCE(:param_1?, column_1),
  column_2 = COALESCE(:param_2?, column_2)
WHERE id = some_id;
```

Currently if I don't pass values to param_1 and param_2 tinypg will throw a `Missing expected key` error.

**Comment by slifty** [comment]
I have another use case for this!  I'm select statements with optional filters (e.g. optional search terms, or forieng key relationships, etc)

At this point I'm forced to pass a placeholder value, but I would much rather have the ability to pass null / undefined and have tinypg insert `NULL`.

Is there any interest in supporting this type of feature?

**Comment by slifty** [comment]
One more use case -- I'd like to have a select query that can *optionally* take limit / offset values.

Ideally I could pass `NULL` to my limit in the event that I don't want to limit in a given instance, so the query would be something like `LIMIT :limit` and accept `null`

**Comment by slifty** [comment]
I'm realizing maybe the name of this issue should be `translate undefined values to NULL` since that would be more flexible and solve the problem directly.

**Comment by dynajoe** [comment]
`translate undefined values to NULL` seems like it could lead to accidental bugs. One benefit of explicitly passing in NULL is that there are no surprises. Perhaps the calling code could substitute `null` where expected?

**Comment by slifty** [comment]
Agreed -- I think when I opened this issue I was under the impression that I could not pass `null` at all.  I doubt that was true!

Being able to pass `null` accomplishes the original need and you're right that translating `undefined` to `null` would be implicit / unexpected behavior that could well lead to bugs.

Thank you!
